frontend/src/components/Candidate-Decider/ApplicantCredentials.tsx

```diff
@@ -12,6 +12,55 @@ type Props = {
   preferredName?: string;
 };
 
+const fixLink = (link: string, git: boolean, linkedIn: boolean): string | undefined => {
+  if (!link) {
+    return undefined;
+  }
+
+  const urlRegex =
+    /(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_+.~#?&//=]*)/g;
+  const matches = link.match(urlRegex);
+
+  let extractedLink = matches ? matches[0].trim() : link.trim();
+
+  let fixedLink: string = '';
+
+  if (!extractedLink.startsWith('https://') && extractedLink.startsWith('http://')) {
+    extractedLink = extractedLink.replace('http://', 'https://');
+  }
+
+  if (extractedLink.includes('https://')) {
+    if (!extractedLink.startsWith('https://')) {
+      const index = extractedLink.indexOf('https://');
+      fixedLink = extractedLink.substring(index);
+    } else {
+      fixedLink = extractedLink;
+    }
+  } else if (linkedIn) {
+    if (extractedLink.startsWith('www.')) {
+      const index = extractedLink.indexOf('https://');
```

Comment:
If it starts with `www.`, then I wouldn't expect the string to have `https://`. Shouldn't we just prefix `extractedLink` with `https://`? Similar questions with the below logic

Reply:
oh wait yeah ill fix that